// router/lib/route.js
const Team = require('../models/Team');

// Get user's team
exports.getUserTeam = async (req, res) => {
  try {
    const team = await Team.findOne({ userId: req.user._id });
    
    if (!team) {
      return res.status(404).json({
        success: false,
        message: 'No team found for this user'
      });
    }
    
    res.status(200).json({
      success: true,
      data: team
    });
  } catch (error) {
    res.status(500).json({
      success: false,
      message: 'Error fetching team',
      error: error.message
    });
  }
};

// Create or update user's team
exports.saveTeam = async (req, res) => {
  try {
    const { teamName, pokemon } = req.body;
    
    // Validate pokemon array
    if (!pokemon || !Array.isArray(pokemon)) {
      return res.status(400).json({
        success: false,
        message: 'Pokemon array is required'
      });
    }
    
    if (pokemon.length > 6) {
      return res.status(400).json({
        success: false,
        message: 'Team cannot have more than 6 Pokemon'
      });
    }
    
    // Validate each pokemon has max 4 moves
    for (let p of pokemon) {
      if (p.selectedMoves && p.selectedMoves.length > 4) {
        return res.status(400).json({
          success: false,
          message: 'Each Pokemon cannot have more than 4 moves'
        });
      }
    }
    
    // Find existing team or create new one
    let team = await Team.findOne({ userId: req.user._id });
    
    if (team) {
      // Update existing team
      team.teamName = teamName || team.teamName;
      team.pokemon = pokemon;
      await team.save();
    } else {
      // Create new team
      team = await Team.create({
        userId: req.user._id,
        teamName: teamName || 'My Team',
        pokemon
      });
    }
    
    res.status(200).json({
      success: true,
      message: 'Team saved successfully',
      data: team
    });
  } catch (error) {
    res.status(500).json({
      success: false,
      message: 'Error saving team',
      error: error.message
    });
  }
};

// Delete user's team
exports.deleteTeam = async (req, res) => {
  try {
    const team = await Team.findOneAndDelete({ userId: req.user._id });
    
    if (!team) {
      return res.status(404).json({
        success: false,
        message: 'No team found to delete'
      });
    }
    
    res.status(200).json({
      success: true,
      message: 'Team deleted successfully'
    });
  } catch (error) {
    res.status(500).json({
      success: false,
      message: 'Error deleting team',
      error: error.message
    });
  }
};

// Clear team (remove all pokemon but keep team)
exports.clearTeam = async (req, res) => {
  try {
    const team = await Team.findOne({ userId: req.user._id });
    
    if (!team) {
      return res.status(404).json({
        success: false,
        message: 'No team found'
      });
    }
    
    team.pokemon = [];
    await team.save();
    
    res.status(200).json({
      success: true,
      message: 'Team cleared successfully',
      data: team
    });
  } catch (error) {
    res.status(500).json({
      success: false,
      message: 'Error clearing team',
      error: error.message
    });
  }
};
